alacritty/src/input.rs

```diff
@@ -616,20 +618,17 @@ impl<'a, T: EventListener, A: ActionContext<T> + 'a> Processor<'a, T, A> {
             _ => (),
         }
 
-        let point = self.ctx.mouse_coords();
-
         // Skip normal mouse events if the message bar has been clicked
-        if let Some(message) = self.message_at_point(point) {
-            // Message should never be `Some` if point is `None`
-            debug_assert!(point.is_some());
-            self.on_message_bar_click(state, point.unwrap(), message, modifiers);
+        if self.message_close_at_cursor() && state == ElementState::Pressed {
+            self.ctx.clear_selection();
+            self.ctx.pop_message();
```

Comment:
You should reset mouse cursor right after closing the button, otherwise you'll still have message modifier.

Reply:
I think we did this before already but I couldn't figure out why. This makes sense, though I don't think I was able to make this happen when testing.

Are you speaking from experience or were you actually able to encounter this bug?

Reply:
I'm speaking from experience here. Just open message bar, press `x` and don't move your mouse. It should stay as `Hand`.